draft-irtf-cfrg-hpke.md

```diff
@@ -866,15 +871,12 @@ For P-256, P-384 and P-521, senders and recipients MUST perform partial
 public-key validation on all public key inputs, as defined in section 5.6.2.3.4
 of {{keyagreement}}. This includes checking that the coordinates are in the
 correct range, that the point is on the curve, and that the point is not the
-point at infinity.
-Additionally, one of the following checks MUST be ensured: the scalar
-given as input to DH is in the interval [1, n-1] where n is the prime
-order of the subgroup; the result of DH is not the point at infinity.
+point at infinity. Additionally, senders and recipients MUST ensure the
+Diffie-Hellman shared secret is not the point at infinity.
 
-For X25519 and X448, validation of public keys
-is not required. Senders and recipients MUST check whether the Diffie-Hellman
-shared secret is the all-zero value and abort if so, as described in
-{{?RFC7748}}.
+For X25519 and X448, validation of public keys is not required. Senders and
```

Comment:
I would rephrase as: "For X25519 and X448, public keys and DH outputs MUST be validated as described in {{RFC7748}}.  In particular, recipients MUST check whether the DH shared secret is the all-zero value and abort if so."

Reply:
👍